### PR TITLE
Strictly check for supported version

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we add a strict check whether the supplied
Elasticsearch version is supported. We check the
`--distribution-version` command line argument for validity but also
the automatically derived distribution version because the distribution
version doesn't have to be specified in all cases.

Relates #1107

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
@elasticmachine test this please

File: esrally/rally.py [hunk 1]
```diff
@@ -56,6 +56,14 @@ def runtime_jdk(v):
             except argparse.ArgumentTypeError:
                 raise argparse.ArgumentTypeError(f"must be a positive number or 'bundled' but was {v}")
 
+    def supported_es_version(v):
```
Comment (dliappis) [inline esrally/rally.py:59]: Reading the definition of [type](https://docs.python.org/3/library/argparse.html#type) it seems to be that using type to do more advanced content based validation like here is a bit of a stretch. (hence the weird `if v:` and `return v`).

I think for now this is ok, but in a future refactoring that we've discussed about offline anyway, we could leverage [action](https://docs.python.org/3/library/argparse.html#action)?

**Review by dliappis — CHANGES_REQUESTED** [review]
This seems to work fine!

Let's remove the unused dot file (AFAICS it's unused) and then it's ready for merging.

File: .es-version.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+6.0.0
```
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply .es-version.txt:1]: That was a leftover. I'll remove.

File: esrally/es-version.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+6.0.0
```
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply esrally/es-version.txt]: Addressed in https://github.com/elastic/rally/pull/1245/commits/e1d7e1978e6a99f3d685a17862c478ff601c20e6.

File: esrally/rally.py (hunk 1 above)
Comment (danielmitterdorfer) [reply esrally/rally.py:59]: I was applying the same logic as for other validations but you have a valid point. I can raise a follow-up issue.

**Comment by danielmitterdorfer** [comment]
@dliappis thanks for your review. I've addressed your comments. Can you please have another look?

**Review by dliappis — APPROVED** [review]
Thanks for iterating. LGTM.

File: esrally/rally.py (hunk 1 above)
Comment (danielmitterdorfer) [reply esrally/rally.py:59]: I've raised https://github.com/elastic/rally/issues/1249 so we won't forget.

**Pull request merged**